vespa/deployment.py

```diff
@@ -440,65 +452,149 @@ def __init__(
     ) -> None:
         """
         Deploy application to the Vespa Cloud (cloud.vespa.ai)
+        There are several ways to initialize VespaCloud:
+        The choices are:
+        - Application source: From python-defined application package or from disk folder.
+        - Control plane access: With api-key (must be added to Vespa Cloud Console) or access token, obtained by interactive login.
+        - Data plane access: mTLS is used by default, but Vespa applications can also be configured to use token based authentication. (token must be added to Vespa Cloud Console, and corresponding auth_token_id must be provided)
+
+        Below are some examples of how to initialize VespaCloud.
+
+        Example usage::
+
+            # 1. Initialize VespaCloud with application package and existing api-key for control plane access.
+            vespa_cloud = VespaCloud(
+                tenant="my-tenant",
+                application="my-application",
+                application_package=app_package,
+                key_location="/path/to/private-key.pem",
+            )
+
+            # 2. Initialize VespaCloud from disk folder by interactive control plane auth.
+            vespa_cloud = VespaCloud(
+                tenant="my-tenant",
+                application="my-application",
+                disk_folder="/path/to/application",
+            )
+
+            # 3. Initialize VespaCloud with application package and token based data plane access.
+            vespa_cloud = VespaCloud(
+                tenant="my-tenant",
+                application="my-application",
+                application_package=app_package,
+                auth_client_token_id="my-token-id", # Must be added in Vespa Cloud Console
+            )
+
 
         :param tenant: Tenant name registered in the Vespa Cloud.
         :param application: Application name in the Vespa Cloud.
-        :param application_package: ApplicationPackage to be deployed.
+        :param application_package: ApplicationPackage to be deployed. Either this or disk_folder must be set.
         :param key_location: Location of the control plane key used for signing HTTP requests to the Vespa Cloud.
         :param key_content: Content of the control plane key used for signing HTTP requests to the Vespa Cloud. Use only when
             key file is not available.
         :param auth_client_token_id: Use token based data plane authentication. This is the token name configured in the Vespa Cloud Console.
-            This is used to configure Vespa services.xml. The token is given read and write permissions.
+            This is used to configure Vespa services.xml. The token is given read and write permissions. If initiliazing from disk_folder, make sure
+            that services.xml is configured to use the provided token_id.
         :param output_file: Output file to write output messages. Default is sys.stdout
+        :param disk_folder: Disk folder for application root.
```

Comment:
I think `application_path` or `application_package_path` would be a better name. It's more in line with what we use already.